[openwrt-24.10] rockchip: fix eMMC corruption on NanoPC-T6 with A3A444 chips

dzikus opened the pull request:
Some NanoPC-T6 boards with A3A444 eMMC chips experience I/O errors and corruption when using HS400 mode. Downgrade to HS200 mode to ensure stable operation.

Fixes: #18844

hauke commented:
Please add this to the main branch first. You can then later backport it to 24.10.

Commit:
rockchip: fix eMMC corruption on NanoPC-T6 with A3A444 chips

Some NanoPC-T6 boards with A3A444 eMMC chips experience I/O errors and
corruption when using HS400 mode. Downgrade to HS200 mode to ensure
stable operation.

Fixes: #18844
Signed-off-by: Grzegorz Sterniczuk <grzegorz@sternicz.uk>
Link: https://github.com/openwrt/openwrt/pull/19398
[Add patch header]
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
(cherry picked from commit c4586ceae85f7f8eefa08c71d0ef67bb07b5e0cc)

dzikus commented:
> Please add this to the main branch first. You can then later backport it to 24.10.

Done https://github.com/openwrt/openwrt/pull/19398

dzikus commented:
> You need to cherry pick -x <sha_commit_from_main>

Done with `make target/linux/refresh`

hauke commented:
A similar change was applied to 24.10 in c77f2b442e32b1f6bc3194143585efc968833e0c.
